src/ui/components/address-action/TransactionSimulation/TransactionSimulation.tsx

```diff
@@ -94,8 +99,13 @@ export function TransactionSimulation({
     useErrorBoundary: true,
   });
 
+  const network = chain ? networks?.getNetworkByName(chain) : null;
   const txInterpretQuery = useQuery({
-    enabled: FEATURE_PAYMASTER_ENABLED ? paymasterEligible === false : true,
+    enabled: FEATURE_PAYMASTER_ENABLED
+      ? network?.supports_sponsored_transactions
```

Comment:
This definitely looks like something that can be moved to a helper. Because we use this sophisticated condition twice :)